pkg/controller.v1/common/pod.go

```diff
@@ -437,10 +477,15 @@ func (jc *JobController) createNewPod(job interface{}, rt, index string, spec *a
 		} else {
 			podTemplate.Spec.SchedulerName = gangSchedulerName
 		}
+
+		if podTemplate.Annotations == nil {
+			podTemplate.Annotations = map[string]string{}
+		}
+		podTemplate.Annotations[gangSchedulingPodGroupAnnotation] = metaObject.GetName()
```

Comment:
i am not sure we need to have gangscheduling here, since we might do not need gangscheduling.

Reply:
@merlintang 

Yeah. it's up to the controller 

You can disable that when you create job controller
https://github.com/kubeflow/xgboost-operator/blob/master/pkg/controller/xgboostjob/xgboostjob_controller.go#L131

Here, I have a minor improvement. Instead of create annotation for all operators, I check configuration first and only create this annotation if operator enable it.